documentation/src/main/asciidoc/elasticsearch-integration.asciidoc

```diff
@@ -286,11 +286,208 @@ public static class Tweet {
+    private String englishTweet;
+
+    @Field
+    @Analyzer(impl = WhitespaceAnalyzer.class) // Elasticsearch built-in "whitespace" analyzer
+    private String whitespaceTweet;
+
+}
+--
+
+===== Custom analyzers
+
+You can also define analyzers within your Hibernate Search mapping using the `@AnalyzerDef` annotation,
+like you would <<section-named-analyzers,do with an embedded Lucene instance>>.
+When Hibernate Search creates the Elasticsearch indexes, the relevant definitions will then be automatically added as a
+https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-custom-analyzer.html[custom analyzer]
+in  the index settings.
+
+[WARNING]
+====
+Analyzer definitions will only be created automatically when Hibernate Search creates the index.
+Hibernate Search will never update analyzer definitions on an existing index
+(no support for the `MERGE` <<elasticsearch-schema-management-strategy,schema management strategy>>)
+and will never validate that pre-existing definitions actually match your Hibernate Search mapping
+(no support for the `VALIDATE` <<elasticsearch-schema-management-strategy,schema management strategy>>).
+====
+
+Two different approaches allow you to define your analyzers with Elasticsearch.
+
+The first, recommended approach is to use the factories provided by the `hibernate-search-elasticsearch` module:
+
+ * `org.hibernate.search.elasticsearch.analyzer.ElasticsearchCharFilterFactory`
+ * `org.hibernate.search.elasticsearch.analyzer.ElasticsearchTokenFilterFactory`
+ * `org.hibernate.search.elasticsearch.analyzer.ElasticsearchTokenizerFactory`
+
+Those classes can be passed to the `factory` attribute of
+the `@CharFilterDef`, `@TokenFilterDef` and `@TokenizerDef` annotations.
+
+The `params` attribute may be used to define the `type` parameter and any other parameter
+accepted by Elasticsearch for this type.
+
+The parameter values will be interpreted as JSON. The parser is not strict, though:
+
+ * quotes around strings may be left out in some cases, as when a string only contains letters.
+ * when quotes *are* required (e.g. your string may be interpreted as a number, and you don't want that), you may use single quotes.
+
+[NOTE]
+====
+You may use the `name` attribute of the `@CharFilterDef`, `@TokenFilterDef` and `@TokenizerDef` annotations
+to define the exact name to give to that definition in the Elasticsearch settings.
+====
+
+[source,java]
+.Example of mapping that defines analyzers on Elasticsearch using the `Elasticsearch*Factory` types
+--
+@Entity
+@Indexed(index = "tweet")
+@AnalyzerDefs(
+	@AnalyzerDef(
+		name = "tweet_analyzer",
+		charFilters = {
+			@CharFilterDef(
+				name = "custom_html_strip",
+				factory = ElasticsearchCharFilterFactory.class,
+				params = {
+					@Parameter(name = "type", value = "'html_strip'"),
+					// One can use Json arrays
+					@Parameter(name = "escaped_tags", value = "['br' 'p']")
```

Comment:
I know the parser is lenient but wouldn't it be better with a , between the array elements?